src/audio/module_adapter/module_adapter.c

```diff
@@ -123,6 +123,17 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 
 	dev->state = COMP_STATE_READY;
 
+#if CONFIG_IPC_MAJOR_4
+	struct sof_ipc_stream_params params = {{ 0 }};
+
+	/* set the stream params based on the module base cfg */
```

Comment:
This is quite whacky looking code to pass a NULL params struct. I would add even more explanation to explain that in IPC3, params is used, but in IPC4 is not used, but instead it will be filled from basecfg.

Reply:
@kv2019i @ranj063 well, not sure if it's important or intentional, but it isn't the same as what is done now. Now these parameters are initialised to 0 only **once** for the entire pipeline. So when traversing the pipeline in `ipc4_comp_params()` if the first component sets some values in it, the next component will get those values. Let me "request changes" for this PR for now to make sure we think this through

Reply:
@lyakh have a look at the updated PR now. I think Ive fixed the issue with missing comp_verify_params(). I've got almost all platforms passing except for LNL HDA(which doesnt seem like an issue with my PR)

Reply:
of course the QB tests as failing @lgirdwood who can help me with that. I cant even access the link